src/main/java/de/tum/in/www1/artemis/repository/CourseRepository.java

```diff
@@ -283,28 +283,54 @@ GROUP BY SUBSTRING(CAST(s.submissionDate AS string), 1, 10), p.student.login
     List<StatisticsEntry> getActiveStudents(@Param("exerciseIds") Set<Long> exerciseIds, @Param("startDate") ZonedDateTime startDate, @Param("endDate") ZonedDateTime endDate);
 
     /**
-     * Fetches the courses to display for the management overview
+     * Get all courses that are not ended yet or have no end date
      *
-     * @param now        ZonedDateTime of the current time. If an end date is set only courses before this time are returned. May be null to return all
-     * @param isAdmin    whether the user to fetch the courses for is an admin (which gets all courses)
-     * @param userGroups the user groups of the user to fetch the courses for (ignored if the user is an admin)
-     * @return a list of courses for the overview
+     * @param now the current time
+     * @return a list of courses that are not ended yet
+     */
+    @Query("""
+            SELECT c
+            FROM Course c
+            WHERE c.endDate IS NULL
+                OR c.endDate >= CAST(:now AS timestamp)
```

Comment:
Is `CAST(:now AS timestamp)` really needed here? We have quite a few places where it does not seem necessary and where we simply use `:now` which is much simpler and easier to understand

Reply:
Indeed unnecessary. We already use it without throughout the system so we know that it works without. I removed the cast.